source/core.py

```diff
@@ -560,6 +560,11 @@ def run(self):
 	else:
 		log.debug("initializing updateCheck")
 		updateCheck.initialize()
+	# If running from source, try to disconnect from the console we may have been executed in.
+	# NVDA may reconnect to read it later,
+	# but it is better to assume we are not connected to anything at the start.
+	if ctypes.windll.kernel32.FreeConsole() != 0:
+		log.debug("Freed from parent console process.")

```

Comment:
Why is this change necessary or related to the rest of the PR?

Reply:
If running NvDA from source in a console with runnvda.bat (in other words python) (not runnvdaw.bat or pythonw) NVDA is associated with the controle, and standard out / standard error are printed there. However, as a screen reader, NVDA itself connects to the console with focus in order to access its contents (at least the older implementation that doesn't use UIA). However, as NVDA is already connected to the console when it starts, then the winConsoleHandler code fails to connect again and or manage connecting / disconnecting from the console with focus. Thus, once NVDA is fully started up, and any critical errors would have been written to the console, as a screen reader, it is better to forcefully disconnect, and it may or may not reconnect later in order to read its content. However, I would be happy to remove this commit and address it in a future pr as it is not specifically to do with Python 3.8. However developers will still hit this if executing runnvda.bat.